Service/CronDataService.php

```diff
@@ -0,0 +1,33 @@
+<?php
+declare(strict_types=1);
+
+namespace Phpro\Scheduler\Service;
+
+use Magento\Cron\Model\Schedule;
+use Phpro\Scheduler\Api\Service\ProviderInterface;
+use Phpro\Scheduler\Util\DateTimeConverter;
+use Phpro\Scheduler\Data\Schedule as ScheduleData;
+
+class CronDataService
+{
+    /**
+     * @var ProviderInterface[]
+     */
+    private $providers;
+
+    public function __construct(array $providers)
+    {
+        $this->providers = $providers;
+    }
+
+    public function getCronData(): array
+    {
+        $data = [];
+
+        foreach ($this->providers as $provider) {
+            $data[] = $provider->provide();
+        }
+
+        return array_merge([], ...$data);
```

Comment:
What is the benefit of doing this? You could as well return the $data array directly